packages/ui/src/core/autocomplete/autocomplete.component.tsx

```diff
@@ -266,18 +285,22 @@ const Autocomplete = intrinsicComponent<AutocompleteProps, HTMLDivElement>(
       return null;
     };
 
-    const renderMenuItem = (item: string, index: number, id: number): any => (
-      <MenuItem
-        key={id >= 0 ? id : index}
-        value={item}
-        noOptionsText={item === noOptionsText}
-        disabled={getOptionDisabled && getOptionDisabled(item, index)}
-        active={(multiple && selected.includes(item)) || item === selected || index === currentItemIndex}
-        onClick={(event: React.MouseEvent<HTMLDivElement>) => handleMenuItemClick(event, item, index)}
-      >
-        {item}
-      </MenuItem>
-    );
+    const renderMenuItem = (item: string, index: number): any => {
+      const optionIndex = getOptionIndex(item);
```

Comment:
Why we need to use getOptionIndex to get the index, we are getting the right index from the parameter that comes from the object.values down there